src/Symfony/Component/DependencyInjection/Compiler/ResolveAutowireInlineAttributesPass.php

```diff
@@ -89,19 +84,14 @@ private function registerAutowireInlineAttributes(\ReflectionFunctionAbstract $m
         if ($method->isVariadic()) {
             array_pop($parameters);
         }
-        $dummyContainer = new ContainerBuilder($this->container->getParameterBag());
+        $paramResolverContainer = new ContainerBuilder($this->container->getParameterBag());
 
         foreach ($parameters as $index => $parameter) {
             if ($isChildDefinition) {
                 $index = 'index_'.$index;
```

Comment:
Renaming of indexes for child definitions is what caused the issue.
The patch fixes this using named arguments instead, which means using Definition::setArgument() instead of replaceArgument() when resolving child definitions.